### PR TITLE
Page resize observer & Toolbar/PageHeaderTools responsiveness based on page width

### PR DESCRIPTION
**Pull request by jschuler**
**What**: Towards https://github.com/patternfly/patternfly-react/issues/6030

- Added a page resize observer to the Page component
- Adds **width**, and **getBreakpoint** to the `PageContext`. The width is the width of the Page (pf-c-page) container. Children components can use the context to base logic around that (like the Toolbar components in this PR). The **getBreakpoint** function defaults to the one in `util.ts`, but can be overwritten if desired.
- Adds a `pf-m-breakpoint-[default|sm|md|lg|xl|2xl]` class as a sibling to the Page (pf-c-page) container. This allows for CSS to target page based breakpoints [as described in this issue](https://github.com/patternfly/patternfly/issues/4622)

A new util function `getBreakpoint` was added to return the breakpoint for a given width.

The `util.ts` **formatBreakpointMods** function was updated to accept a `breakpoint` argument.
Before, you would have a string of class names that are media query based, i.e. `pf-m-hidden pf-m-visible-on-md pf-m-hidden-on-lg`, or `pf-m-spacer-none pf-m-spacer-md-on-md`, etc. If the `breakpoint` argument is supplied, we instead return now a single class based on the known width.

To test this, I added a lightbulb icon button to the masthead which toggles a drawer in this demo:
https://patternfly-react-pr-6827.surge.sh/components/page/react-demos/sticky-section-group/

**Additional issues/PRs**: https://github.com/patternfly/patternfly/issues/4622, https://github.com/patternfly/patternfly/pull/4625
**TBD**: More PRs to follow to update components that use the **formatBreakpointMods** function, or set breakpoint based modifier classes

Demo (note, this also contains the change from core here https://github.com/patternfly/patternfly/pull/4625 which is not yet merged):

https://user-images.githubusercontent.com/869106/150210012-f693ab20-2565-4470-bff1-fbcd81622862.mov

### CODE REVIEW

File: packages/react-core/src/components/Page/Page.tsx [hunk 1]
```diff
@@ -110,15 +135,20 @@ export class Page extends React.Component<PageProps, PageState> {
...
-    const { isManagedSidebar, onPageResize } = this.props;
-    if (isManagedSidebar || onPageResize) {
+    const { isManagedSidebar, onPageResize, useResizeObserver } = this.props;
+    if (isManagedSidebar || onPageResize || useResizeObserver) {
       if (canUseDOM) {
-        window.addEventListener('resize', this.handleResize);
+        if (useResizeObserver) {
+          this.observer = getResizeObserver(this.pageRef.current, this.handleResize);
+        } else {
+          window.addEventListener('resize', this.handleResize);
```
Comment (dlabrecq) [inline packages/react-core/src/components/Page/Page.tsx:150]: Curious why we offer `useResizeObserver` as a choice? 

Note that `getResizeObserver` will use a window event listener if `containerRefElement` is not available.

Comment (jschuler) [reply packages/react-core/src/components/Page/Page.tsx:150]: hmm, I suppose we can just enable the observer by default without providing a prop. And will update the code to remove the extra listener, thanks!

**Review by dlabrecq — APPROVED** [review]

File: packages/react-core/src/helpers/util.ts [hunk 2]
```diff
@@ -262,32 +262,84 @@ export const setBreakpointCssVars = (
...
-    '2xl'?: string;
-    '3xl'?: string;
-  },
+  mods: Mods,
   styles: any,
-  stylePrefix: string = ''
-) =>
-  Object.entries(mods || {})
+  stylePrefix: string = '',
+  breakpoint?: 'xs' | 'sm' | 'md' | 'lg' | 'xl' | '2xl'
```
Comment (mattnolting) [inline packages/react-core/src/helpers/util.ts:285]: My only feedback, I'd like to see `xs` -> `default` until we have an actual value for `xs`.

**Review by mattnolting — APPROVED** [review]
Fantastic work @jschuler! L🤖 TM!

**Review by mattnolting — APPROVED** [review]
👏👏👏  Love it

**Review by tlabaj — APPROVED** [review]
LGTM

**Review by nicolethoen — COMMENTED** [review]
@jschuler I think that at least the stacked toolbar example is behaving differently after this update. Can you just verify that the [toolbar examples](https://patternfly-react-pr-6827.surge.sh/components/toolbar) are all behaving as you'd expect with the updates?

The stacked example is now collapsing the ToolbarToggleGroup at full page width - it didn't do that before.

File: packages/react-core/src/helpers/util.ts [hunk 3]
```diff
@@ -318,7 +318,7 @@ export const formatBreakpointMods = (
  */
 export const getBreakpoint = (width: number): 'default' | 'sm' | 'md' | 'lg' | 'xl' | '2xl' => {
   if (width === null) {
-    width = canUseDOM ? window.innerWidth : 1200;
+    return null;
```
Comment (nicolethoen) [inline packages/react-core/src/helpers/util.ts:321]: This does cause a classname to get applied - `pf-m-breakpoint-null` so I feel like that's not quite ideal?

Comment (jschuler) [reply packages/react-core/src/helpers/util.ts:321]: thanks, I added a check in the page component to make sure width is defined first

**Review by nicolethoen — APPROVED** [review]

**Pull request merged**